x/nodes/keeper/reward.go

```diff
@@ -10,21 +10,38 @@ import (
 
 // RewardForRelays - Award coins to an address (will be called at the beginning of the next block)
 func (k Keeper) RewardForRelays(ctx sdk.Ctx, relays sdk.BigInt, address sdk.Address) sdk.BigInt {
+	// feature flags
+	isAfterRSCAL := k.Cdc.IsAfterNamedFeatureActivationHeight(ctx.BlockHeight(), codec.RSCALKey)
+	isAfterNonCustodial := k.Cdc.IsAfterNonCustodialUpgrade(ctx.BlockHeight())
+
+	// The conditions of the original non-custodial issue:
+	isDuringNonCustodialIssue := isAfterNonCustodial && isAfterRSCAL && ctx.BlockHeight() <= codec.NonCustodial1RollbackHeight
+
 	// get validator
 	validator, found := k.GetValidator(ctx, address)
 	if !found {
 		ctx.Logger().Error(fmt.Errorf("no validator found for address %s; at height %d\n", address.String(), ctx.BlockHeight()).Error())
 		return sdk.ZeroInt()
 	}
 
-	if k.Cdc.IsAfterNonCustodialUpgrade(ctx.BlockHeight()) {
+	if isAfterNonCustodial {
 		address = k.GetOutputAddressFromValidator(validator)
 	}
 
+	// This simulates the issue that happened between the original non-custodial rollout and the rollback height
+	// This code is needed to allow a 'history replay' of the issue during sync-from-scratch
```

Comment:
Not actionable: good learning lesson for myself on the importance of blockchain history & immutability.